src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,174 @@
+    private double currentTime = 0;
+    private double lastTime = 0;
+
+    private Shooter() {
+        configureMotor();
+        linearSystemLoop = configStateSpace(true);
+        if (Robot.isSimulation()) {
+            simCollection = motor.getSimCollection();
+        }
+    }
+
+    /**
+     * Gets the single instance of the shooter.
+     *
+     * @return shooter instance.
+     */
+    public static Shooter getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Shooter();
+        }
+        return INSTANCE;
+    }
+
+    private void configureMotor() {
+        motor.configAllSettings(getConfiguration());
+        motor.setInverted(INVERSION_TYPE);
+    }
+
+    /**
+     * State space configuration function. Note that there are 2 different configurations.
+     *
+     * @param isInertiaBased is the configuration for the state space.
+     *                       If the value is true, the state space is based off of an inertia model.
+     *                       Otherwise, the state space is based off of a voltage equation model.
+     * @return the linear system loop (based on the type).
+     */
+    private LinearSystemLoop<N1, N1, N1> configStateSpace(boolean isInertiaBased) {
+        final DCMotor motor = DCMotor.getFalcon500(1);
+
+        LinearSystem<N1, N1, N1> flywheel_plant;
+        if (isInertiaBased) {
+            flywheel_plant = LinearSystemId.createFlywheelSystem(motor, J, GEAR_RATIO);
+        } else {
+            flywheel_plant = LinearSystemId.identifyVelocitySystem(Kv, Ka);
+        }
+        if (Robot.isSimulation()) {
+            flywheelSim = new FlywheelSim(
+                    flywheel_plant,
+                    motor,
+                    GEAR_RATIO);
+        }
+
+        LinearQuadraticRegulator<N1, N1, N1> quadraticRegulator = new LinearQuadraticRegulator<>(
+                flywheel_plant,
+                VecBuilder.fill(VELOCITY_TOLERANCE),
+                VecBuilder.fill(COST_LQR),
+                LOOP_PERIOD);
+        quadraticRegulator.latencyCompensate(flywheel_plant, LOOP_PERIOD, Units.millisecondsToSeconds(TALON_TIMEOUT));
+
+        KalmanFilter<N1, N1, N1> kalmanFilter = new KalmanFilter<>(
+                Nat.N1(), Nat.N1(),
+                flywheel_plant,
+                VecBuilder.fill(MODEL_TOLERANCE),
+                VecBuilder.fill(SENSOR_TOLERANCE),
+                LOOP_PERIOD);
+
+        return new LinearSystemLoop<>(
+                flywheel_plant,
+                quadraticRegulator,
+                kalmanFilter,
+                NOMINAL_VOLTAGE, LOOP_PERIOD);
+    }
+
+    /**
+     * Calculates the velocity setpoint according to the distance from the target.
+     * Once the data from the shooter is acquired this function will be changed.
+     *
+     * @param distance is the distance from the target. [m]
+     * @return 15. [rpm]
+     */
+    public static double getSetpointVelocity(double distance) {
```

Comment:
Continuing the conversation from [here](https://github.com/Galaxia5987/robot-2022/pull/49#discussion_r795997098):
> The function will be based off of the values that we measure with our testing. This objectively means that the interpolation belongs to the Shooter subsystem itself, and not the command. (@littlebro123)

I'm not sure I understand. Why does it matter where do the values come from? Would it be different if they would come from, say,  theoretical calculations?

Reply:
I agree that it makes more sense to have this method in the command, shooter can be seen as a module that receives a velocity and gets to it to shoot. The command is what decides that velocity and the command receives the distance.